flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -765,7 +765,7 @@ public void dropPartition(ObjectPath tablePath, CatalogPartitionSpec partitionSp
 
 	@Override
 	public List<CatalogPartitionSpec> listPartitions(ObjectPath tablePath, CatalogPartitionSpec partitionSpec)
-			throws TableNotExistException, TableNotPartitionedException, CatalogException {
+			throws TableNotExistException, TableNotPartitionedException, PartitionNotExistException, CatalogException {
```

Comment:
In case of invalid partition spec, we should throw `PartitionSpecInvalidException`